staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -993,6 +993,18 @@ func RunTestList(ctx context.Context, t *testing.T, store storage.Interface, ign
 			rvMatch:     metav1.ResourceVersionMatchNotOlderThan,
 			expectedOut: []example.Pod{*preset[0], *preset[1], *preset[2], *preset[3], *preset[4]},
 		},
+		{
+			name:   "verify list returns updated version of object; filter returns one item, page size equal to total list size with current resource version and match=NotOlderThan",
+			prefix: "/pods",
+			pred: storage.SelectionPredicate{
+				Field: fields.OneTermEqualSelector("spec.nodeName", "fakeNode"),
+				Label: labels.Everything(),
+				Limit: 5,
+			},
+			rv:          list.ResourceVersion,
+			rvMatch:     metav1.ResourceVersionMatchNotOlderThan,
+			expectedOut: []example.Pod{*preset[0]},
+		},
```

Comment:
This is a good test - I like it, but it's not exactly what the original test was testing.

That test was testing if some object existed before but was deleted, it will no longer be returned.
I think we want such test to still exist.

We probably want in seedMultiLevelData add another object `baz` (in second namespace?) and have it first created as everything else, and then Delete it explicitly so that it no longer exists.

Reply:
@wojtek-t - done. I've retained the update test case and added the delete test case additionally.